app/models/dunning_campaign.rb

```diff
@@ -4,7 +4,10 @@ class DunningCampaign < ApplicationRecord
   include PaperTrailTraceable
 
   belongs_to :organization
+
   has_many :thresholds, class_name: "DunningCampaignThreshold", dependent: :destroy
+  has_many :customers, foreign_key: :applied_dunning_campaign_id, dependent: :nullify
```

Comment:
I'm not 100% sure about setting nullify as dependent strategy here... it makes sense but we also need to reset the customer dunning campaign attempts counter to zero and nullify the last attempt run at timestamp....